tasks/cheryl-birthday/src/main.rs

```diff
@@ -0,0 +1,141 @@
+// This version is based on the Go version on Rosettacode
+
+#[derive(PartialEq, Debug, Copy, Clone)]
+enum Month {
+    May,
+    June,
+    July,
+    August,
+}
+
+#[derive(PartialEq, Debug, Copy, Clone)]
+struct Birthday {
+    month: Month,
+    day: u8,
+}
+
+impl Birthday {
+    fn month_unique_in(&self, birthdays: &[Birthday]) -> bool {
+        birthdays
+            .iter()
+            .filter(|birthday| birthday.month == self.month)
+            .count()
+            == 1
+    }
+
+    fn day_unique_in(&self, birthdays: &[Birthday]) -> bool {
+        birthdays
+            .iter()
+            .filter(|birthday| birthday.day == self.day)
+            .count()
+            == 1
+    }
+
+    fn month_with_unique_day_in(&self, birthdays: &[Birthday]) -> bool {
+        birthdays
+            .iter()
+            .any(|birthday| self.month == birthday.month && birthday.day_unique_in(birthdays))
+    }
+}
+
+fn solution() -> Option<Birthday> {
+    let choices: Vec<Birthday> = vec![
+        Birthday {
+            month: Month::May,
+            day: 15,
+        },
+        Birthday {
+            month: Month::May,
+            day: 16,
+        },
+        Birthday {
+            month: Month::May,
+            day: 19,
+        },
+        Birthday {
+            month: Month::June,
+            day: 17,
+        },
+        Birthday {
+            month: Month::June,
+            day: 18,
+        },
+        Birthday {
+            month: Month::July,
+            day: 14,
+        },
+        Birthday {
+            month: Month::July,
+            day: 16,
+        },
+        Birthday {
+            month: Month::August,
+            day: 14,
+        },
+        Birthday {
+            month: Month::August,
+            day: 15,
+        },
+        Birthday {
+            month: Month::August,
+            day: 17,
+        },
+    ];
+
+    // Albert knows the month but doesn't know the day.
+    // So the month can't be unique within the choices.
+    let filtered = choices
```

Comment:
I think using [`Vec::retain`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain) for each filter would be a bit better than collecting into new `Vec`s each time.

```rust
let choices_copy = choices.clone();
choices.retain(|birthday| !birthday.month_unique_in(&choices_copy));
```